include/gadget/sockets-map.h

```diff
@@ -100,42 +100,63 @@ gadget_socket_lookup(const struct __sk_buff *skb)
 	int l4_off;
 	__u16 h_proto;
 	int i;
+	long err;
 
 	key.netns = skb->cb[0]; // cb[0] initialized by dispatcher.bpf.c
-	h_proto = load_half(skb, offsetof(struct ethhdr, h_proto));
+	err = bpf_skb_load_bytes(skb, offsetof(struct ethhdr, h_proto),
+				 &h_proto, sizeof(h_proto));
+	if (err < 0)
+		return 0;
+	h_proto = bpf_ntohs(h_proto);
```

Comment:
nit: Can you put this on the case statements to perform this operation at compilation time on the constants instead of at runtime? (I don't know if the compiler does it for us)